mirte_robot/robot.py

```diff
@@ -45,28 +98,68 @@ def __init__(self):
         self.begin_time = time.time()
         self.last_call = 0
 
-        # Call /stop and /start service to disable/enable the ROS diff_drive_controller
-        # By default this class will control the rbot though PWM (controller stopped). Only in case
+        # Call controller_manager/switch_controller service to disable/enable the ROS diff_drive_controller
+        # By default this class will control the robot though PWM (controller stopped). Only in case
         # the controller is needed, it will be enabled.
-        self.stop_controller_service = rospy.ServiceProxy('stop', Empty, persistent=True)
-        self.start_controller_service = rospy.ServiceProxy('start', Empty, persistent=True)
+        self.switch_controller_service = self._node.create_client(
+            SwitchController, "controller_manager/switch_controller"
+        )
+
+        if ROS_DISTRO[0] >= "i":  # Check if the ROS Distro is IRON or newer
+            # Not available untill ROS Iron
+            if not self._node.wait_for_node(
+                self._hardware_namespace + "telemetrix", 10
+            ):
+                self._node.get_logger().fatal(
+                    f"Telemetrix node at '{self._node.get_namespace() + self._hardware_namespace  + 'telemetrix'}' was not found! Aborting"
+                )
+                exit(-1)
+        list_parameters: rclpy.client.Client = self._node.create_client(
```

Comment:
You currently seem to call the list_parameters on every type (eg motor, distance etc). I think it would be more efficient to just get all the params once.

Reply:
In my current way of doing, it actually filters the parameters.
It would slightly more efficient if it is called once, but then parsing gets more complex.
An optimization that would be straightforward to do is to call all services in the beginning, so we do not have to wait as long (since they are already async).